adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/AmqpContext.java

```diff
@@ -63,21 +63,21 @@ ProtonDelivery delivery() {
     }
 
     /**
-     * Gets the tenant identifier of this context's resource.
+     * Gets the tenant identifier for this context.
      *
      * @return The tenant identifier.
      */
     String getTenantId() {
-        return resource.getTenantId();
+        return isDeviceAuthenticated() ? authenticatedDevice.getTenantId() : resource.getTenantId();
     }
 
     /**
-     * Gets the device identifier of this context's resource.
+     * Gets the device identifier for this context.
      *
      * @return The device identifier.
      */
     String getDeviceId() {
-        return resource.getResourceId();
+        return isDeviceAuthenticated() ? authenticatedDevice.getDeviceId() : resource.getResourceId();
```

Comment:
For a gateway connecting to the adapter, this will probably not return the intended value. A gateway usually will publish data *on behalf of* another device (of the same tenant), therefore the device ID from the link's target address will usually **not** be the same as the device ID of the authenticated device.

Reply:
humm.. ok, i will have to think about this a little bit